modules/develop/pages/data-transforms/how-transforms-work.adoc

```diff
@@ -0,0 +1,47 @@
+= How Data Transforms Work
+:description: Learn how Redpanda data transforms work.
+
+Redpanda provides the framework to build and deploy inline transformations on data written to Redpanda topics, delivering processed and validated data to consumers in the format they expect. Redpanda does this directly inside the broker, eliminating the need to manage a separate stream processing environment or use 3rd-party tools. 
+
+image::shared:wasm1.png[Data transforms in a broker] 
+
+To learn how to build and deploy data transforms, see xref:./run-transforms.adoc[].
+
+== WebAssemby in Redpanda
+
+Data transforms use a WebAssembly (Wasm) engine inside a Redpanda broker, allowing Redpanda to control the entire transform lifecycle. For example, Redpanda can stop and start transforms when partitions are moved or to free up system resources for other tasks. 
+
+Within Repdanda, there's a single Raft controller that manages cluster information, including data transforms. On every shard (core), Redpanda knows what data transforms exist in the cluster, as well as metadata about the transform, such as input and output topics and environment variables. 
+
+Wasm is executed within a runtime in the broker. Data transforms take an input topic and map it to an output topic, running a virtual machine (VM) on the same shard as every partition leader for a topic. Redpanda uses just-in-time (JIT) compilation, which compiles the function in memory, writes it to executable space, then runs the directly translated machine code. When a shard becomes the leader of a given partition on the input topic, Redpanda does the following:
+
+. Spins up a Wasm VM using the JIT-compiled WebAssembly module
+. Pushes records from the input partition into the Wasm VM
+. Writes the output, which may go to the same shard on the same machine, or to a different shard on the same machine, or to a different machine
+
+image::shared:wasm_architecture.png[Wasm architecture in Redpanda]
+
+The transform function processes every record produced to an input topic and returns zero or more records that are then produced to an output topic. Data transforms are applied to all partitions on an input topic. A record is processed after it has been successfully written to disk on the input topic. Because the transform happens in the background after the write finishes, the transform doesn't affect the original produced record and doesn't block writes to the input topic. 
+
+image::shared:wasm_flow.png[Flow of record transform]
+
+A new data transform processes the input topic from the latest offset. When a partition loses leadership, the broker issues an internal notification, and Redpanda shuts down the transform on that core. When another broker gains leadership of the partition, it also issues a notification, and the transform subsystem starts a processor and resumes from the last committed offset. This is why transforms have at-least-once semantics. 
+
+== Configure Wasm properties
```

Comment:
@micheleRP as you indicated in the PR comments we should move the actual configs to the 'Run Wasm Transforms' page under 'Configure Data Transforms memory' (after the 'enable data transforms' step where we set the flag).

Here we seem to have a   'Data Transforms resource management' or maybesection that explains how resource management works without the explicit conf (perhaps linking over to the conf on the other page once we get to that part).  We should probably replace the how-to conf part here with a couple more sentences about memory management / reservations etc without referring to the conf and then  forklift the part starting with the conf properties to the other page.

cc @rockwotj